File: 0_correlation_analysis.py
import pandas as pd

# Read the CSV data into a DataFrame
data = pd.read_csv('your_data.csv')

# Calculate the correlation matrix
correlation_matrix = data.corr()

# Get the absolute correlation values for each variable pair
abs_correlation = correlation_matrix.abs()

# Find the pairs with high correlation
high_correlation_pairs = (abs_correlation > 0.7) & (abs_correlation < 1.0)
high_correlation_variables = abs_correlation[high_correlation_pairs].stack().reset_index()
high_correlation_variables.columns = ['Variable 1', 'Variable 2', 'Correlation']

# Sort the pairs by correlation strength
sorted_pairs = high_correlation_variables.sort_values('Correlation', ascending=False)

# Display the strongly correlated pairs
print("Strongly Correlated Variable Pairs:")
print(sorted_pairs)


'''In both examples, we read the CSV data into a DataFrame (or data frame in R), calculate the correlation matrix using the corr function, and obtain the absolute correlation values for each variable pair. We then filter for pairs with correlation values above a certain threshold (in this case, > 0.7 and < 1.0) and sort the pairs based on the correlation strength.

The output will provide you with the strongly correlated variable pairs, along with their correlation coefficients. By examining these pairs, you can identify the most interesting variables that exhibit strong correlations.'''

File: 0_correlation_analysis.r
# Read the CSV data into a DataFrame
data <- read.csv("your_data.csv")

# Calculate the correlation matrix
correlation_matrix <- cor(data)

# Get the absolute correlation values for each variable pair
abs_correlation <- abs(correlation_matrix)

# Find the pairs with high correlation
high_correlation_pairs <- (abs_correlation > 0.7) & (abs_correlation < 1.0)
high_correlation_variables <- subset(as.data.frame(which(high_correlation_pairs, arr.ind = TRUE), stringsAsFactors = FALSE), select = -c(row))

# Rename the columns
colnames(high_correlation_variables) <- c("Variable 1", "Variable 2")

# Add the correlation values
high_correlation_variables$Correlation <- abs_correlation[high_correlation_pairs]

# Sort the pairs by correlation strength
sorted_pairs <- high_correlation_variables[order(high_correlation_variables$Correlation, decreasing = TRUE), ]

# Display the strongly correlated pairs
cat("Strongly Correlated Variable Pairs:\n")
print(sorted_pairs)
